bob-backend/src/pearl/disk_controller.rs

```diff
@@ -538,6 +538,22 @@ impl DiskController {
         cnt
     }
 
+    pub(crate) async fn corrupted_blobs_count(&self) -> usize {
+        if *self.state.read().await == GroupsState::Ready {
+            let mut cnt: usize = 0;
+            for group in self.groups.read().await.iter() {
+                let holders_guard = group.holders();
+                let holders = holders_guard.read().await;
+                for holder in holders.iter() {
```

Comment:
It is not ok to access `holders` on this level. It breaks SOLID principles. You should add `corrupted_blobs_count` function to the `Group` struct and iterate holders on the group level.

I also see that you have copied the code structure from the `blobs_count` function, so I ask you to also update `blobs_count` and `index_memory` functions